### PR TITLE
Cybercom updates

### PR DESCRIPTION
**Pull request by mbstacy**
Cybercom Updates:

1. Docs merged - Need to update system configuration
2. Mongo options(distinct/aggregation)
3. data_store Safe Methods

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -9,7 +9,7 @@ djangorestframework-simplejwt==4.7.2; python_version > '3.6'
 djangorestframework-xml==2.0.0
 djangorestframework-yaml==2.0.0
 PyJWT==2.1.0
-pymongo==3.12.0
+pymongo==3.7.2
```
Comment (tdpearson) [inline requirements.txt:12]: Was there an error with the newer version of pymongo?

File: cybercom_queue/views.py [hunk 2]
```diff
@@ -134,8 +134,10 @@ def post(self, request, task_name=None, format=None):
         if type(kwargs) is not dict:
             return Response({'error': 'kwargs must be a JSON object'})
         tags = request.data.get('tags', [])
+        user_info = request.user.__class__.objects.filter(pk=request.user.id).values().first()
+        user_info.pop('password', None)
```
Comment (tdpearson) [inline cybercom_queue/views.py:138]: Does this have the potential to leak credentials?

File: dc_config/images/celery/requirements.txt [hunk 3]
```diff
@@ -1,3 +1,2 @@
 celery==5.1.2
-pymongo==3.12.0
-
+pymongo==3.7.2
```
Comment (tdpearson) [inline dc_config/images/celery/requirements.txt:2]: Did the newer version of pymongo throw any errros?

File: docs/pages/configuration.md [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+       - config/ssl/backend/client/key.pem
+       - config/ssl/backend/client/cert.pem
+       - config/ssl/backend/server/key.pem
+       - config/ssl/backend/server/cert.pem
+       - config/ssl/testca/cacert.pem
+
+  1. LetsEncrypt location:
+     * NGINX
+       - config/ssl/nginx/keys/dhparam.pem
+       - config/ssl/nginx/letcencrypt/etc/live/*
```
Comment (tdpearson) [inline docs/pages/configuration.md:47]: This is out of date. I will take a closer look and get this section updated.

File: dc_config/images/celery/requirements.txt (hunk 3 above)
Comment (mbstacy) [reply dc_config/images/celery/requirements.txt:2]: I updated when changing the insert_many and Insert_one code. No errors on my side. I needed to upgrade to 3.2 but thought why not go to the current stable version.

File: docs/pages/configuration.md (hunk 4 above)
Comment (mbstacy) [reply docs/pages/configuration.md:47]: Yes, I need you to figure out with your current configuration what is the appropriate docs. Or totally remove?

File: docs/pages/configuration.md [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+# Uncomment and configure to enable send email
+EMAIL_BACKEND = 'django.core.mail.backends.smtp.EmailBackend'
+EMAIL_HOST = 'smtp.gmail.com'
+EMAIL_PORT = 587
+EMAIL_HOST_USER = 'username@gmail.com'
+EMAIL_HOST_PASSWORD = 'password'
+EMAIL_USE_TLS = True
+~~~
+
+* To enable Admin email notifications, add a line similar to the following to the config/api_config.py file. *Refer to Django's [documentation](https://docs.djangoproject.com/en/1.8/topics/email/) for more details on configuring email.*
```
Comment (tdpearson) [inline docs/pages/configuration.md:62]: Location and settings have moved to secrets env file underdc_config. I will update this section.

File: docs/pages/configuration.md [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+  ~~~
+  #Rabbitmq
+  echo "************** Rabbitmq        ***********"
+  docker run -d -p 5671:5671 --name example_rabbitmq \
+  ~~~
+1. Update firewall rules on host to allow access to these ports
+
+### Turn Off Debug Mode for RESTful API
+1. Set DEBUG = False in config/api_config.py
+1. Add host(s) to ALLOWED_HOSTS list. See Django's documentation on the [ALLOWED_HOSTS](https://docs.djangoproject.com/en/1.8/ref/settings/#std:setting-ALLOWED_HOSTS) setting for more detail.
```
Comment (tdpearson) [inline docs/pages/configuration.md:98]: This is out of date. I will take a closer look and updated.

File: docs/pages/remote_worker.md [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+* PIP - [Install](https://packaging.python.org/install_requirements_linux/#installing-pip-setuptools-wheel-with-linux-package-managers)
+* Copies of client certificates and credentials to communicate with central cyberCommons server:
+  - MongoDB
+    - config/ssl/backend/client/mongodb.pem
+    - config/ssl/testca/cacert.pem
+  - RabbitMQ
+    - config/ssl/backend/client/key.pem
+    - config/ssl/backend/client/cert.pem
+    - config/ssl/testca/cacert.pem
+* Open RabbitMQ and MongoDB ports on the cyberCommons server
```
Comment (tdpearson) [inline docs/pages/remote_worker.md:19]: Paths to certs have changed. I will take a closer look and update.

File: docs/pages/remote_worker.md [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+
+1. Install Python's virtualenv
+
+        $ pip install virtualenv
+
+2. Create virtual environment and activate
+
+
+        $ virtualenv virtpy
+        $ source virtpy/bin/activate
```
Comment (tdpearson) [inline docs/pages/remote_worker.md:31]: These are Python 2.7 specific. I will update to use Python 3 conventions.

File: cybercom_queue/views.py (hunk 2 above)
Comment (mbstacy) [reply cybercom_queue/views.py:138]: I do not believe that it does.... I'm removing the password and passing through encrypted communication to rabbitmq--> celery task. I'm open to any thoughts or updated code. The request user does not have the plain text password but wanted to remove any security vulnerabilities.

File: requirements.txt (hunk 1 above)
Comment (mbstacy) [reply requirements.txt:12]: No

File: docs/pages/configuration.md (hunk 6 above)
Comment (mbstacy) [reply docs/pages/configuration.md:98]: Can delete if not needed. Just did not know if you have time to update?

File: requirements.txt (hunk 1 above)
Comment (mbstacy) [reply requirements.txt:12]: I also tried MONGO_VERSION=5.0.3 and no problems. I know that is a breaking change with older data stored in mongo. I reverted back to your  MONGO_VERSION=4.4.6

**Review by tdpearson — COMMENTED** [review]
I will test these changes and address the items I flagged to review for myself. Please provide answers to the other comments.

File: requirements.txt (hunk 1 above)
Comment (tdpearson) [reply requirements.txt:12]: This is specific to pymongo. I will test on my end with both versions of pymongo and mongodb.

File: dc_config/images/celery/requirements.txt (hunk 3 above)
Comment (tdpearson) [reply dc_config/images/celery/requirements.txt:2]: I will test and confirm on my end. I agree to use the latest version if possible.

File: docs/pages/configuration.md (hunk 4 above)
Comment (tdpearson) [reply docs/pages/configuration.md:47]: Good point. I will make changes and get feedback before doing the PR upstream.

File: requirements.txt (hunk 1 above)
Comment (tdpearson) [reply requirements.txt:12]: Both versions of pymongo and mongodb have passed my testing. I am for bumping these to their latest versions. For those that need to support version 4.x of MongoDB, that can be done by downgrading the version in dc_config/cybercom_config.env. Any objections to this?

Comment (mbstacy) [reply requirements.txt:12]: Agree, No Objections! I wanted to make sure your current volume would work with the new changes. I guess that is really easy to remove volume and `make init` again.

**Pull request merged**